Mixpanel/Mixpanel.m

```diff
@@ -49,10 +49,33 @@ + (Mixpanel *)sharedInstanceWithToken:(NSString *)apiToken
 
 + (Mixpanel *)sharedInstance
 {
-    if (sharedInstance == nil) {
-        MPLogWarning(@"sharedInstance called before sharedInstanceWithToken:");
+    if (!instances || instances.count == 0) {
+        MPLogWarning(@"sharedInstance called before creating a Mixpanel instance");
+        return nil;
+    }
+
+    if (instances.count > 1) {
+        MPLogWarning(@"sharedInstance can only be called when you have one Mixpanel instance");
+        return nil;
+    }
+
+    return [instances objectForKey:[[instances keyEnumerator] nextObject]];
+}
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        self.eventsQueue = [NSMutableArray array];
+        self.peopleQueue = [NSMutableArray array];
+        self.timedEvents = [NSMutableDictionary dictionary];
+        self.shownSurveyCollections = [NSMutableSet set];
+        self.shownNotifications = [NSMutableSet set];
+        if (!instances) {
```

Comment:
maybe put this in `dispatch_once` instead? it's effectively as fast as checking nil but also gets thread safety for free (i read an interesting article i could probably find if you want more details on why)